src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -134,6 +134,23 @@ def store_task_res(self, task_res: TaskRes) -> Optional[UUID]:
                 )
                 return None
 
+            # Limit the TaskRes TTL to not exceed the
+            # expiration time of the TaskIns it replies to.
+            # Condition: TaskIns.created_at + TaskIns.ttl ≥
+            #            TaskRes.created_at + TaskRes.ttl
+            max_allowed_ttl = (
+                task_ins.task.created_at + task_ins.task.ttl - task_res.task.created_at
+            )
+            if task_res.task.ttl > max_allowed_ttl:
+                log(
+                    WARNING,
+                    "Received TaskRes with TTL %s exceeding the allowed maximum TTL %s."
+                    "Updating TTL to the allowed maximum.",
+                    task_res.task.ttl,
+                    max_allowed_ttl,
+                )
+                task_res.task.ttl = max_allowed_ttl
```

Comment:
Instead of updating the TTL, we should fail. It the provided TTL is wrong here, it's likely due to a client-side bug that should be addressed.